jormungandr-scenario-tests/src/node.rs

```diff
@@ -15,43 +26,162 @@ error_chain! {
         CannotSpawnNode {
             description("Cannot spawn the node"),
         }
+
+        InvalidHeaderHash {
+            description("Invalid header hash"),
+        }
+
+        InvalidBlock {
+            description("Invalid block"),
+        }
+
+        NodeStopped (status: Status) {
+            description("the node is no longer running"),
+        }
     }
 }
 
+pub enum NodeBlock0 {
+    Hash(HeaderHash),
+    File(PathBuf),
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum Status {
+    Running,
+    Failure,
+    Exit(ExitStatus),
+}
+
+#[derive(Clone)]
+struct ProgressBarController {
+    progress_bar: ProgressBar,
+    prefix: String,
+}
+
+pub struct NodeController {
+    alias: NodeAlias,
+    settings: NodeSetting,
+    progress_bar: ProgressBarController,
+    status: Arc<Mutex<Status>>,
+}
+
 pub struct Node {
     alias: NodeAlias,
 
     #[allow(unused)]
     temp_dir: Temp,
 
     process: Process,
+
+    progress_bar: ProgressBarController,
+    node_settings: NodeSetting,
+    status: Arc<Mutex<Status>>,
 }
 
 const NODE_CONFIG: &str = "node_config.yaml";
 const NODE_SECRET: &str = "node_secret.yaml";
 
-impl Node {
+impl NodeController {
```

Comment:
Can you please document the purpose of `NodeController` as separate from `Node`?